make/xml.cmake

```diff
@@ -37,3 +37,99 @@ function(XML_LINT)
 
 endfunction(XML_LINT)
 
+function(XML_SORT)
+  # ~~~
+  # XML_SORT(
+  # NAME
+  # FILE
+  # OUTPUT
+  # )
+  set(oneValueArgs NAME FILE OUTPUT)
+  cmake_parse_arguments(
+    XML_SORT
+    ""
+    "${oneValueArgs}"
+    ""
+    ${ARGN}
+    )
+
+  set(XML_SORT_XSL ${symbiflow-arch-defs_SOURCE_DIR}/common/xml/xmlsort.xsl)
+
+  get_file_location(XML_SORT_INPUT_LOCATION ${XML_SORT_FILE})
+
+  get_file_target(XML_SORT_INPUT_TARGET ${XML_SORT_FILE})
+  get_target_property(INCLUDE_FILES ${XML_SORT_INPUT_TARGET} INCLUDE_FILES)
```

Comment:
append_file_dependency does this recursion for you.  Just all append_file_dependency(DEPS ${XML_SORT_FILE})